### PR TITLE
Compare gating version against existing deployment versions

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY
Compare gating version against existing deployment versions and set upgradeFrom status.

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

### CODE REVIEW

File: roles/installer/tasks/check_previous_version.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  register: existing_deployment
+
+- name: Set prevous_version version based on AWX CR version status
+  ansible.builtin.set_fact:
+    prevous_version: "{{ existing_deployment.resources[0].status.version | default(0.0.0) }}"
+  when: existing_deployment['resources'] | length > 0
+
+- name: If prevous_version is less than or equal to gating_version, set upgrade_from to prevous_version
+  ansible.builtin.set_fact:
+    upgrade_from: "{{ prevous_version }}"
```
Comment (rcarrillocruz) [inline roles/installer/tasks/check_previous_version.yml]: upgraded_from

File: roles/installer/tasks/check_previous_version.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    name: "{{ ansible_operator_meta.name }}"
+  register: existing_deployment
+
+- name: Set prevous_version version based on AWX CR version status
+  ansible.builtin.set_fact:
+    prevous_version: "{{ existing_deployment.resources[0].status.version | default(0.0.0) }}"
+  when: existing_deployment['resources'] | length > 0
+
+- name: If prevous_version is less than or equal to gating_version, set upgrade_from to prevous_version
```
Comment (rcarrillocruz) [inline roles/installer/tasks/check_previous_version.yml]: previous

File: roles/installer/tasks/check_previous_version.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  ansible.builtin.set_fact:
+    upgrade_from: "{{ prevous_version }}"
+  when: prevous_version | version_compare(gating_version, '<=')
```
Comment (rcarrillocruz) [inline roles/installer/tasks/check_previous_version.yml]: previous

File: roles/installer/tasks/main.yml [hunk 4]
```diff
@@ -23,6 +23,10 @@
     namespace: "{{ ansible_operator_meta.namespace }}"
   register: awx_web_deployment
 
+- name: Check for existing deployment for previous version
+  include_tasks: previous_version_check.yml
```
Comment (rcarrillocruz) [inline roles/installer/tasks/main.yml]: check_previous_version.yml

Comment (rooftopcellist) [reply roles/installer/tasks/main.yml]: Thanks, I just address all these comments, thanks for the quick feedback!

File: roles/installer/tasks/check_previous_version.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  k8s_info:
+    api_version: "{{ api_version }}"
+    kind: "{{ kind }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    name: "{{ ansible_operator_meta.name }}"
+  register: existing_deployment
+
+- name: Set previous_version version based on AWX CR version status
+  ansible.builtin.set_fact:
+    previous_version: "{{ existing_deployment.resources[0].status.version | default(0.0.0) }}"
```
Comment (dsavineau) [inline roles/installer/tasks/check_previous_version.yml]:
```suggestion
    previous_version: "{{ existing_deployment.resources[0].status.version | default('0.0.0') }}"
```

File: roles/installer/tasks/check_previous_version.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  when: existing_deployment['resources'] | length > 0
+
+- name: If previous_version is less than or equal to gating_version, set upgraded_from to previous_version
+  ansible.builtin.set_fact:
+    upgraded_from: "{{ previous_version }}"
+  when: previous_version | version_compare(gating_version, '<')
```
Comment (dsavineau) [inline roles/installer/tasks/check_previous_version.yml]:
```suggestion
  when: previous_version is version_compare(gating_version, '<')
```

File: roles/installer/tasks/update_status.yml [hunk 7]
```diff
@@ -55,6 +55,15 @@
...
   when: awx_web_pod_name != ''
 
+- name: Update upgradedFrom status
+  operator_sdk.util.k8s_status:
+    api_version: '{{ api_version }}'
+    kind: "{{ kind }}"
+    name: "{{ ansible_operator_meta.name }}"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    status:
+      upgradedFrom: "{{ upgraded_from | default(omit) }}"
```
Comment (dsavineau) [inline roles/installer/tasks/update_status.yml]:
```suggestion
      upgradedFrom: "{{ upgraded_from }}"
  when: upgraded_from is defined
```

**Review by dsavineau — APPROVED** [review]

**Review by rcarrillocruz — APPROVED** [review]

**Pull request merged**